ethers-providers/src/transports/quorum.rs

```diff
@@ -190,21 +191,58 @@ impl<T: JsonRpcClientWrapper> QuorumProvider<T> {
 
         let mut numbers = vec![];
         let mut errors = vec![];
+
+        // 1. First we try to reach quorum
+        // 2. After reaching quorum, rocord how long it took to
+        //    reach quorum
+        // 3. Wait for any remaining responses to be return, equal
+        //    to the amount of time it took to reach quorum.
+        let required_weight = self.quorum_weight();
+        let mut weight_achieved = 0;
+
+        // timestamp of when we started
+        let start = Instant::now();
+
+        // wait until we've reached quorum
         while !queries.is_empty() {
             let (response, _index, remaining) = future::select_all(queries).await;
             queries = remaining;
             match response {
-                Ok(v) => numbers.push(v),
+                Ok(v) => {
+                    weight_achieved += v.0.weight;
+                    numbers.push(v);
+                    if weight_achieved >= required_weight {
+                        break;
+                    }
+                }
                 Err(e) => errors.push(e),
             }
         }
 
+        // Current grace period to wait for remaining requests is
+        // to wait for how long it initially took to reach quorum
+        let quorum_reached_timestamp = Instant::now();
+        let quorum_grace_period = quorum_reached_timestamp.duration_since(start);
+
+        // try and wait for any remaining requests
+        let _ = tokio::time::timeout(quorum_grace_period, async {
+            while !queries.is_empty() {
+                let (response, _index, remaining) = future::select_all(queries).await;
+                queries = remaining;
+                match response {
+                    Ok(v) => numbers.push(v),
+                    Err(e) => errors.push(e),
+                }
+            }
+        })
+        .await;
+
```

Comment:
might be nice to have a trace log showing the `quorum_grace_period` and how many queries got a response / which didn't